### PR TITLE
Sweets/mint with uri

### PR DESCRIPTION
**Pull request by sweetmantech**
# Music Game Metadata Renderer
- set `image` on `purchase`.
- set `animation_url` on `purchase`.
- set `description` on `purchase`.

### Demo
- [0x93350fAEa6dC6B1486d7313b44cA9Eac1927B2C4](https://goerli.etherscan.io/address/0x93350fAEa6dC6B1486d7313b44cA9Eac1927B2C4#readContract)
Screenshot 2023-02-14 at 11 45 27 AM

### CODE REVIEW

File: src/ERC721MusicGame.sol [hunk 1]
```diff
@@ -394,6 +378,21 @@ contract ERC721Drop is
...
+        ) = abi.decode(initialData, (string, string, string));
+        bytes memory data = abi.encode(
+            description,
+            imageURI,
+            animationURI,
+            _lastMintedTokenId()
+        );
+        config.metadataRenderer.initializeWithData(data);
+
         emit IERC721Drop.Sale({
```
Comment (crazzywizard) [inline src/ERC721MusicGame.sol:396]: 🎉 love the use of abi encode here

File: src/metadata/MusicGameMetadataRenderer.sol [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    /// @notice Event for a new edition initialized
+    /// @dev admin function indexer feedback
+    event NewMusicInitialized(
+        address indexed target,
+        uint256 tokenId,
+        string description,
+        string imageURI,
+        string animationURI
+    );
```
Comment (crazzywizard) [inline src/metadata/MusicGameMetadataRenderer.sol:46]: 💭 what does this do? Just curious

File: src/metadata/MusicGameMetadataRenderer.sol [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    ) external requireSenderAdmin(target) {
+        _tokenInfos[target][tokenId].imageURI = imageURI;
+        _tokenInfos[target][tokenId].animationURI = animationURI;
+        emit MediaURIsUpdated({
+            target: target,
+            sender: msg.sender,
+            imageURI: imageURI,
+            animationURI: animationURI
+        });
+    }
```
Comment (crazzywizard) [inline src/metadata/MusicGameMetadataRenderer.sol:87]: is there a case where you would want to update the animationURI or the imageURI alone?

File: test/ERC721DropUpgrades.t.sol [hunk 4]
```diff
@@ -6,15 +6,15 @@ import {DSTest} from "ds-test/test.sol";
...
 import {IERC721Drop} from "../src/interfaces/IERC721Drop.sol";
-import {ERC721Drop} from "../src/ERC721Drop.sol";
+import {ERC721MusicGame} from "../src/ERC721MusicGame.sol";
 import {DummyMetadataRenderer} from "./utils/DummyMetadataRenderer.sol";
 import {MockUser} from "./utils/MockUser.sol";
 import {IMetadataRenderer} from "../src/interfaces/IMetadataRenderer.sol";
 import {FactoryUpgradeGate} from "../src/FactoryUpgradeGate.sol";
 import {ERC721DropProxy} from "../src/ERC721DropProxy.sol";
 
 contract ERC721DropTest is DSTest {
```
Comment (crazzywizard) [inline test/ERC721DropUpgrades.t.sol:16]: :nit should rename test and file to `ERC721MusicGame`

**Review by crazzywizard — APPROVED** [review]
LGTM :shipit:

File: src/metadata/MusicGameMetadataRenderer.sol (hunk 2 above)
Comment (sweetmantech) [reply src/metadata/MusicGameMetadataRenderer.sol:46]: are you asking what `event`s do?

or, this event, in particular?

File: src/metadata/MusicGameMetadataRenderer.sol (hunk 3 above)
Comment (sweetmantech) [reply src/metadata/MusicGameMetadataRenderer.sol:87]: possibly, for now, it's not part of our UX.

However, if we need it, we could always just call this fx setting the values we don't want to change as the same value. ✏️ 

If you think this is important, let's chat about the use-case in our UX and how we plan on using it. 🙏

File: test/ERC721DropUpgrades.t.sol (hunk 4 above)
Comment (sweetmantech) [reply test/ERC721DropUpgrades.t.sol:16]: that test file is below right?

This contract isn't being used because we aren't doing any upgrades.

Comment (sweetmantech) [reply test/ERC721DropUpgrades.t.sol:16]: so, maybe I should delete this? 🚮

**Pull request merged**